### PR TITLE
Manual pages link in source files

### PR DESCRIPTION
**Pull request by Spencer-Kinny**
### Motivation and Context
Appended Comments in source files

### Description
Added following links in below source files
- ./module/avl/avl.c -> https://illumos.org/man/9f/avl
- ./module/nvpair/nvpair.c -> https://illumos.org/man/3lib/libnvpair and https://illumos.org/man/3nvpair/nvlist_alloc
- ./module/os/linux/spl/spl-kstat.c -> https://illumos.org/man/1M/kstat
- ./module/os/freebsd/spl/spl_kstat.c -> https://illumos.org/man/1M/kstat

### Types of changes

- [+] Documentation (a change to man pages or other documentation)

### Checklist:

- [+] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [+] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [+] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by Ornias1993** [comment]
Actually I think those manpages just need to be migrated to the OpenZFS documentation, instead of blindly linking too them. Are you even sure the full Illumnos manpage still applies? Because there has been quite a distance between OpenZFS and Illumnos by now.

**Comment by behlendorf** [comment]
I suggested simply linking to them for now rather than adding them to the source tree.  They should still be correct for the most part, we haven't changed the behavior of these library functions.  Tackling adding proper developer documentation for all of these interfaces would be a bigger effort.  This at least provides some helpful information.

File: module/nvpair/nvpair.c [hunk 1]
```diff
@@ -25,6 +25,12 @@
  * Copyright 2018 RackTop Systems.
  */
 
+/*
+ * Links to Illumos.org for more information on Interface Libraries:
+ * [1] https://illumos.org/man/3lib/libnvpair
```
Comment (freqlabs) [inline module/nvpair/nvpair.c:30]: Might link to the kernel versions of these instead (or additionally):
https://illumos.org/man/9f/nvlist_alloc
https://illumos.org/man/9f/nvlist_next_nvpair
https://illumos.org/man/9f/nvpair_value_byte

Comment (behlendorf) [reply module/nvpair/nvpair.c:30]: Good point.  Yes, we should link to the kernel versions.

Comment (Spencer-Kinny) [reply module/nvpair/nvpair.c:30]: @freqlabs and @behlendorf 
Should I add above three links in "module/nvpair/nvpair.c " as well

Comment (behlendorf) [reply module/nvpair/nvpair.c:30]: Yes, please do.   Similarly we should add `https://illumos.org/man/9f/kstat_create` to the `spl_kstat.c` files.

Comment (Spencer-Kinny) [reply module/nvpair/nvpair.c:30]: okay, done

**Comment by freqlabs** [comment]
@Spencer-Kinny Please rebase and squash your commits on this branch so there is just one commit for the addition of the links.

**Comment by Spencer-Kinny** [comment]
> 
> 
> @Spencer-Kinny Please rebase and squash your commits on this branch so there is just one commit for the addition of the links.

@freqlabs 
While trying to rebase and squash, I accidentally removed the commits from my local branch and when im trying to commit again, it says nothing to commit. could you please help with this

**Comment by freqlabs** [comment]
@Spencer-Kinny no problem, I took care of it for you!

* Rebased
* Squashed commits

**Comment by Spencer-Kinny** [comment]
> 
> 
> @Spencer-Kinny no problem, I took care of it for you!
> 
>     * Rebased
> 
>     * Squashed commits

@freqlabs 
Thank you

**Pull request merged**